bot/exts/fun/latex/_latex_cog.py

```diff
@@ -0,0 +1,103 @@
+import hashlib
+import re
+import string
+from pathlib import Path
+from typing import BinaryIO, Optional
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+
+FORMATTED_CODE_REGEX = re.compile(
+    r"(?P<delim>(?P<block>```)|``?)"        # code delimiter: 1-3 backticks; (?P=block) only matches if it's a block
+    r"(?(block)(?:(?P<lang>[a-z]+)\n)?)"    # if we're in a block, match optional language (only letters plus newline)
+    r"(?:[ \t]*\n)*"                        # any blank (empty or tabs/spaces only) lines before the code
+    r"(?P<code>.*?)"                        # extract all code inside the markup
+    r"\s*"                                  # any more whitespace before the end of the code markup
+    r"(?P=delim)",                          # match the exact same delimiter from the start again
+    re.DOTALL | re.IGNORECASE,              # "." also matches newlines, case insensitive
+)
```

Comment:
Ideally we would use the regex from bot-core for this.
However, it hasn't been installed on sir-lance yet. So if you don't want to implement that as part of this PR it's fine.